framework/packages/abstract-client/src/account.rs

```diff
@@ -0,0 +1,162 @@
+use abstract_core::{
+    manager::{state::AccountInfo, InfoResponse},
+    objects::{gov_type::GovernanceDetails, namespace::Namespace, AssetEntry},
+    version_control::NamespaceResponse,
+};
+use abstract_interface::{
+    Abstract, AbstractAccount, AccountDetails, ManagerQueryFns, RegisteredModule, VCQueryFns,
+};
+use cw_orch::contract::Contract;
+use cw_orch::prelude::*;
+use serde::Serialize;
+
+use crate::{
+    application::Application, client::AbstractClientResult, infrastructure::Infrastructure,
+};
+
+pub struct AccountBuilder<'a, Chain: CwEnv> {
+    pub(crate) abstr: &'a Abstract<Chain>,
+    name: Option<String>,
+    description: Option<String>,
+    link: Option<String>,
+    namespace: Option<String>,
+    base_asset: Option<AssetEntry>,
+    // TODO: Decide if we want to abstract this as well.
+    governance_details: Option<GovernanceDetails<String>>,
+    // TODO: How to handle install_modules?
+}
+
+impl<'a, Chain: CwEnv> AccountBuilder<'a, Chain> {
+    pub(crate) fn new(abstr: &'a Abstract<Chain>) -> Self {
+        Self {
+            abstr,
+            name: None,
+            description: None,
+            link: None,
+            namespace: None,
+            base_asset: None,
+            governance_details: None,
+        }
+    }
+
+    pub fn name(self, name: impl Into<String>) -> Self {
+        Self {
+            name: Some(name.into()),
+            ..self
+        }
+    }
+
+    pub fn description(self, description: impl Into<String>) -> Self {
+        Self {
+            description: Some(description.into()),
+            ..self
+        }
+    }
+
+    pub fn link(self, link: impl Into<String>) -> Self {
+        Self {
+            link: Some(link.into()),
+            ..self
+        }
+    }
+
+    pub fn namespace(self, namespace: impl Into<String>) -> Self {
+        Self {
+            namespace: Some(namespace.into()),
+            ..self
+        }
+    }
+
+    pub fn base_asset(self, base_asset: AssetEntry) -> Self {
+        Self {
+            base_asset: Some(base_asset),
+            ..self
+        }
+    }
+
+    pub fn governance_details(self, governance_details: GovernanceDetails<String>) -> Self {
+        Self {
+            governance_details: Some(governance_details),
+            ..self
+        }
+    }
+
+    pub fn build(self) -> AbstractClientResult<Account<Chain>> {
```

Comment:
Whenever possible it's recommended to make the builder function itself take `&self` because then you can easily create multiple accounts with the same / slightly different settings.

Reply:
Ah yeah that makes sense, will change.